core/src/main/scala/kafka/server/ClientQuotaManager.scala

```diff
@@ -155,50 +165,62 @@ class ClientQuotaManager(private val config: ClientQuotaManagerConfig,
   def quotasEnabled: Boolean = quotaTypesEnabled != QuotaTypes.NoQuotas
 
   /**
-   * Records that a user/clientId changed some metric being throttled (produced/consumed bytes, request processing time etc.)
-   * If quota has been violated, callback is invoked after a delay, otherwise the callback is invoked immediately.
-   * Throttle time calculation may be overridden by sub-classes.
-   * @param sanitizedUser user principal of client
-   * @param clientId clientId that produced/fetched the data
-   * @param value amount of data in bytes or request processing time as a percentage
-   * @param callback Callback function. This will be triggered immediately if quota is not violated.
-   *                 If there is a quota violation, this callback will be triggered after a delay
-   * @return Number of milliseconds to delay the response in case of Quota violation.
-   *         Zero otherwise
-   */
-  def maybeRecordAndThrottle(sanitizedUser: String, clientId: String, value: Double, callback: Int => Unit): Int = {
-    if (quotasEnabled) {
-      val clientSensors = getOrCreateQuotaSensors(sanitizedUser, clientId)
-      recordAndThrottleOnQuotaViolation(clientSensors, value, callback)
-    } else {
-      // Don't record any metrics if quotas are not enabled at any level
-      val throttleTimeMs = 0
-      callback(throttleTimeMs)
-      throttleTimeMs
-    }
+    * Records that a user/clientId changed some metric being throttled (produced/consumed bytes, request processing
+    * time, etc.) If quota has been violated, return throttle time in milliseconds. Throttle time calculation may be
+    * overridden by sub-classes.
+    * @param request client request
+    * @param value amount of data in bytes or request processing time as a percentage
+    * @return throttle time in milliseconds
+    */
+  def maybeRecord(request: RequestChannel.Request, value: Double): Int = {
```

Comment:
I think the word `maybe` in the original method name `maybeRecordAndThrottle` is mostly for `throttle`. It is probably more intuitive to name this method `getThrottleTimeMs`.

Reply:
Based on your next comment, I think it still makes sense to call this maybeRecord() (or maybeRecordAndGetThrottleTimeMs?) because recording only happens when quotasEnabled == true. And also rename maybeThrottle() to throttle(). I made those changes. What do you think?